astropy/units/tests/test_quantity_ufuncs.py

```diff
@@ -18,9 +20,47 @@
 from astropy.utils.compat.numpycompat import NUMPY_LT_1_25
 from astropy.utils.compat.optional_deps import HAS_SCIPY
 
-testcase = namedtuple("testcase", ["f", "q_in", "q_out"])
-testexc = namedtuple("testexc", ["f", "q_in", "exc", "msg"])
-testwarn = namedtuple("testwarn", ["f", "q_in", "wfilter"])
+
+class testcase(NamedTuple):
+    """A test case for a ufunc."""
+
+    f: Callable
+    """The ufunc to test."""
+
+    q_in: tuple[Quantity]
+    """The input quantities."""
+
+    q_out: tuple[Quantity]
+    """The expected output quantities."""
+
+
+class testexc(NamedTuple):
+    """A test case for a ufunc that should raise an exception."""
+
+    f: Callable
+    """The ufunc to test."""
+
+    q_in: tuple[Quantity]
+    """The input quantities."""
+
+    exc: type
+    """The expected exception type."""
+
+    msg: str | None
+    """The expected exception message."""
+
+
+class testwarn(NamedTuple):
+    """A test case for a ufunc that should raise a warning."""
+
+    f: Callable
+    """The ufunc to test."""
+
+    q_in: tuple[Quantity]
+    """The input quantities."""
+
+    wfilter: str
+    """The expected warning filter."""
```

Comment:
@mhvk, these can be a lot shorter if I don't include docstrings. Do you think they are useful?

Reply:
If we truly don't care about documentation for these test helpers we can switch these back to `namedtuple`.

Reply:
This part of the tests is rather different than everything else, and my tendency would be to change it; for now, I'd just let them be.